lib/ansible/module_utils/aws/elbv2.py

```diff
@@ -720,7 +720,7 @@ def _compare_condition(self, current_conditions, condition):
                         current_condition['SourceIpConfig']['Values'][0] == condition['SourceIpConfig']['Values'][0]):
                     condition_found = True
                     break
-            elif current_condition['Field'] == condition['Field'] and current_condition['Values'] == condition['Values']:
+            elif current_condition['Field'] == condition['Field'] and current_condition['Values'].sort() == condition['Values'].sort():
```

Comment:
You need to use `.sort()` first and then use it, since `.sort()` returns None and will always results True in comparison.

```
>>> a = [1, 3, 2]
>>> b = [1]
>>> a.sort() == b.sort()
True
>>> a
[1, 2, 3]
>>> b
[1]
>>> a == b
False
```

Reply:
Alternatively: use `sorted()`, i.e.
```suggestion
            elif current_condition['Field'] == condition['Field'] and sorted(current_condition['Values']) == sorted(condition['Values']):
```